From 7b1ad037d92201aee894afd9cc70a852721f2d9d Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Thu, 24 May 2018 14:11:45 +0200
Subject: [PATCH 4/8] redhat: define pseries-rhel7.6.0 machine types

RH-Author: Laurent Vivier <lvivier@redhat.com>
Message-id: <20180524141145.7532-3-lvivier@redhat.com>
Patchwork-id: 80470
O-Subject: [RHEL-7.6 qemu-kvm-rhev PATCH 2/2] redhat: define pseries-rhel7.6.0 machine types
Bugzilla: 1557054
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/ppc/spapr.c | 26 ++++++++++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index 6f005a0..e2df370 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -4354,19 +4354,41 @@ DEFINE_SPAPR_MACHINE(2_1, "2.1", false);
 #endif
 
 /*
+ * pseries-rhel7.6.0
+ */
+
+static void spapr_machine_rhel760_instance_options(MachineState *machine)
+{
+}
+
+static void spapr_machine_rhel760_class_options(MachineClass *mc)
+{
+    /* Defaults for the latest behaviour inherited from the base class */
+}
+
+DEFINE_SPAPR_MACHINE(rhel760, "rhel7.6.0", true);
+
+/*
  * pseries-rhel7.5.0
+ * like SPAPR_COMPAT_2_11 and SPAPR_COMPAT_2_10
+ * SPAPR_CAP_HTM already enabled in 7.4
+ *
  */
+#define SPAPR_COMPAT_RHEL7_5                                           \
+    HW_COMPAT_RHEL7_5                                                  \
 
 static void spapr_machine_rhel750_instance_options(MachineState *machine)
 {
+    spapr_machine_rhel760_instance_options(machine);
 }
 
 static void spapr_machine_rhel750_class_options(MachineClass *mc)
 {
-    /* Defaults for the latest behaviour inherited from the base class */
+    spapr_machine_rhel760_class_options(mc);
+    SET_MACHINE_COMPAT(mc, SPAPR_COMPAT_RHEL7_5);
 }
 
-DEFINE_SPAPR_MACHINE(rhel750, "rhel7.5.0", true);
+DEFINE_SPAPR_MACHINE(rhel750, "rhel7.5.0", false);
 
 /*
  * pseries-rhel7.5.0-sxxm
-- 
1.8.3.1

